### ISSUE TITLE
New mlogloss eval metric crash

### ISSUE THREAD

**Issue by pommedeterresautee** [dmlc/xgboost#204]
```R
data(agaricus.train, package='xgboost')
dtrain <- xgb.DMatrix(agaricus.train$data, label = agaricus.train$label)
dtest <- dtrain
watchlist <- list(eval = dtest, train = dtrain)
param <- list(max.depth = 2, eta = 1, silent = 1)

#Error message (as expected)
bst <- xgb.train(param, dtrain, nthread = 2, nround = 2, watchlist, eval_metric = "fdsg")

#Works
bst <- xgb.train(param, dtrain, nthread = 2, nround = 2, watchlist, eval_metric = "logloss")

#Crash
bst <- xgb.train(param, dtrain, nthread = 2, nround = 2, watchlist, eval_metric = "mlogloss")
```

**Comment by pommedeterresautee** [comment]
Related to commit https://github.com/dmlc/xgboost/commit/e1538ae615896d1527895a8a972b522f24389de1

**Comment by pommedeterresautee** [comment]
New metric is for multiclass learning and the example is binary learning. 
May be a check in C code may help?

**Comment by hetong007** [comment]
Hi @pommedeterresautee , can you observe crash consistently in `mlogloss`? And can you observe anything from other metrics? Thanks! I am also experiencing a similar issue.

**Comment by pommedeterresautee** [comment]
Yes each time I run xgboost with mlogloss it crashes. I checked merror as it is a multiclass metric like mlogloss and merror doesn't crash.

**Comment by tqchen** [comment]
mlogloss was only for multicass case ie softmax and softprob. They do not
work with binary classification

Tianqi

**Comment by tqchen** [comment]
I have made a fix on this so that the error won't cause the crash, please check if it works for you

**Comment by pommedeterresautee** [comment]
I know it s closed (no time to try before), but I just test the fix anyway :-)
Fix works, no crash, just error message.

Kind regards,
Michael
